### PR TITLE
Linting pattern fixes

### PR DESCRIPTION
**Pull request by vikr01**
- Fixed all linting glob patterns to match zero or more times instead of one time. `.(foo|bar)` would now be ~~`.*(foo|bar)`~~ `.{foo,bar}` ([followed glob patterns](https://github.com/isaacs/node-glob/blob/master/README.md#glob-primer))

~~- Removed .prettierrc in favor of using the `--parser json` flag~~

- Added `stylelint --fix` to lint-staged for css & scss

~~- Added `lint-json-fix` script~~

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -34,13 +36,22 @@
...
       "git add"
     ],
-    "*.(json|css|scss|md)|.(babelrc|prettierrc|eslintrc)": [
-      "prettier --ignore-path .eslintignore --write",
+    "*.json|.*(babelrc|eslintrc|stylelintrc)": [
+      "prettier --ignore-path .eslintignore --parser json --write",
+      "git add"
+    ],
+    "*.*(css|scss)": [
+      "stylelint --ignore-path .eslintignore --syntax scss --fix",
```
Comment (amilajack) [inline package.json]: How come we're using two css formatters here? Why not one? Does `stylelint` account for cases that `prettier` doesn't?

File: .prettierrc [hunk 2]
```diff
@@ -1,11 +0,0 @@
-{
```
Comment (amilajack) [inline .prettierrc]: Why was this removed?

File: package.json [hunk 3]
```diff
@@ -14,14 +14,16 @@
     "flow-typed": "rimraf flow-typed/npm && flow-typed install --overwrite || true",
     "lint": "cross-env NODE_ENV=development eslint --cache --format=node_modules/eslint-formatter-pretty .",
     "lint-fix": "yarn --silent lint --fix; exit 0",
+    "lint-json-fix": "prettier --ignore-path .eslintignore --parser json --write '**/*(*.json|.*(babelrc|eslintrc|stylelintrc))'",
```
Comment (amilajack) [inline package.json]: Why did we add a new script for this instead of reuse an existing script?

File: .prettierrc (hunk 2 above)
Comment (vikr01) [reply .prettierrc]: I added this file before to recognize .babelrc, .eslintrc, and .prettierrc as json filetypes. But adding the flag `--parser json` is better.

File: package.json (hunk 3 above)
Comment (vikr01) [reply package.json]: Add to lint-fix?

File: package.json (hunk 1 above)
Comment (vikr01) [reply package.json]: `stylelint` caught a missing generic font error, which `prettier` couldn't.
`stylelint` seems to be the equivalent of `eslint` for styles, so I think it's better to keep it.

File: package.json (hunk 3 above)
Comment (vikr01) [reply package.json]: Could make postlint-fix into this, but it would be long.
`"postlint-fix": "concurrently \"prettier --ignore-path .eslintignore --single-quote --write '**/*.*(js|jsx)'\" \"prettier --ignore-path .eslintignore --parser json --write '**/*(*.json|.*(babelrc|eslintrc|stylelintrc))'\""`

Comment (vikr01) [reply package.json]: Could make postlint-fix into this, but it would be a bit long.
`concurrently \"prettier --ignore-path .eslintignore --single-quote --write '**/*.*(js|jsx)'\" \"prettier --ignore-path .eslintignore --parser json --write '**/*(*.json|.*(babelrc|eslintrc|stylelintrc))'\"`

Comment (amilajack) [reply package.json]: One of the issues with the boilerplate is that is has way too many npm scripts. Try combining this with the `lint-fix` script

Comment (vikr01) [reply package.json]: Merged into `postlint-fix`. This required `.prettierrc` to be added back to avoid the `--parser json` flag (since it would cause errors for js/jsx).

The problem with doing this though, is that if a user wanted to add another filetype to be included in the lint they need to include it in both .prettierrc and package.json.

Comment (vikr01) [reply package.json]: Also used the more commonly used glob patterns, since it reduces the length of the command. To be consistent, I also had to change it in the lint-staged config and `postlint-styles-fix`.

(i.e. `*.*(js|jsx)` becomes `*.{js,jsx}`)

**Comment by amilajack** [comment]
Good stuff. Also can you investigate the flow issue that is causing CI to fail

**Comment by vikr01** [comment]
PR'd  #1712.
